storage/sqlite/src/lib.rs

```diff
@@ -283,19 +286,28 @@ impl Sqlite {
     }
 }
 
+/// Implements in memory database, useful for testing
+impl Default for Sqlite {
```

Comment:
This is wrong.

We have a new() function that doesn't support in-memory storage, and a default trait that does that. This seems too abrupt.

Please remove the default, and modify the new() function to support both.

Reply:
Or leave `new` to open a database from file and add a separate `new_inmemory` constructor.

Reply:
I added a `new_in_memory` in `Sqlite` that does that.

However the `wallet/storage/src/internal/mod.rs` contains the following:

```rust
impl<B: Default + storage::Backend> Store<B> {
    /// Create a default storage (mostly for testing, may want to remove this later)
    pub fn new_empty() -> crate::Result<Self> {
        Self::new(B::default())
    }
}
```

I am not sure how I can create a new "in memory" database without parametrising over `Sqlite`. Alternatively I can move this impl to the test.rs file.

Reply:
Default has to go... it doesn't make any sense.

Reply:
I don't think we need the new_empty() function for the wallet. You can remove that.